privaterelay/apps.py

```diff
@@ -10,10 +13,80 @@
 ROOT_DIR = os.path.abspath(os.curdir)
 
 
+class VersionInfo(TypedDict):
+    source: str
+    version: str
+    commit: str
+    build: str
+
+
+@cache
+def get_version_info(base_dir: str | Path | None = None) -> VersionInfo:
+    """Return version information written by build process."""
+    if base_dir is None:
+        base_path = Path(settings.BASE_DIR)
+    else:
+        base_path = Path(base_dir)
+    version_json_path = base_path / "version.json"
+    info = {}
+    if version_json_path.exists():
+        with version_json_path.open() as version_file:
+            try:
+                info = json.load(version_file)
+            except ValueError:
+                pass
+            if not hasattr(info, "get"):
+                info = {}
+    version_info = VersionInfo(
+        source=info.get("source", "https://github.com/mozilla/fx-private-relay"),
+        version=info.get("version", "unknown"),
+        commit=info.get("commit", "unknown"),
+        build=info.get("build", "not built"),
+    )
+    return version_info
+
+
+def get_profiler_startup_data() -> Tuple[str | None, str | None, str | None]:
+    if settings.RELAY_CHANNEL not in ("dev", "stage", "prod"):
+        return (None, None, None)
+
+    if settings.RELAY_CHANNEL in ("dev", "stage"):
+        service = "fxprivaterelay-nonprod"
+        project_id = "moz-fx-fxprivate-nonprod-6df0"
+    if settings.RELAY_CHANNEL == "prod":
+        service = "fxprivaterelay"
+        project_id = "moz-fx-fxprivaterel-prod-6700"
+
+    version_info = get_version_info()
+    version = version_info.get("version", "unknown")
+
+    return service, version, project_id
+
+
 class PrivateRelayConfig(AppConfig):
     name = "privaterelay"
 
     def ready(self) -> None:
+        # Set up Google Cloud Profiler
+        service, version, project_id = get_profiler_startup_data()
+        if service != None:
+            import googlecloudprofiler
+
+            # Make sure the expect gcp_key.json file exists
+            gcp_key_json_path = Path(settings.BASE_DIR) / "gcp_key.json"
+            if gcp_key_json_path.exists():
+                with gcp_key_json_path.open() as gcp_key_file:
+                    try:
+                        # Make sure the expect gcp_key.json file is valid json
+                        json.load(gcp_key_file)
+                        googlecloudprofiler.start(
+                            service=service,
+                            service_version=version,
+                            project_id=project_id,
+                        )
+                    except ValueError:
+                        print(f"error during json.load({gcp_key_file})")
```

Comment:
Nit: should the error mention the path instead of the text stream?

```suggestion
                        print(f"error during json.load({gcp_key_json_path})")
```